### ISSUE TITLE
ENH: support full range of NumPy ufunc ops

### ISSUE THREAD

**Issue by tylerjereddy** [kokkos/pykokkos#27]
My initial testing is actually looking pretty positive for `pykokkos` providing fairly-easy access to feature-parity with the NumPy `ufuncs`. One case where it looks like we currently would need to write manual code is for `np.sign` (see below). I wonder if, given that we can use things like `exp` and `sqrt` as-is "just fine" with `pykokkos` (which is great!) if we might benefit from making a convenience wrapper for `sign` (and any other simple missing functions) as well?

```python
@pk.workload
class SqrtView1D:
    def __init__(self, threads: int, view: pk.View1D[pk.double]):
        self.threads: int = threads
        self.view: pk.View1D[pk.double] = view

    @pk.main
    def run(self) -> None:
        pk.parallel_for(self.threads, self.pfor)

    @pk.workunit
    def pfor(self, tid: int) -> None:
        self.view[tid] = sign(self.view[tid])


def test_1d_ufunc_vs_numpy():
    view: pk.View1d = pk.View([10], pk.double)
    view[:] = np.arange(10, dtype=np.float64)
    sqrt_kokkos_instance = SqrtView1D(threads=10, view=view)
    pk.execute(pk.ExecutionSpace.Default, sqrt_kokkos_instance)
    actual = sqrt_kokkos_instance.view
    expected = np.sign(range(10))
    assert_allclose(actual, expected)
```

```
File "/vast/home/treddy/github_projects/pykokkos/tests/test_views.py", line 335:
 undefined symbol sign used
```

**Comment by NaderAlAwar** [comment]
Yeah this would be great. The reason we support the current math functions (like `exp` and `sqrt`) is because they can be trivially mapped to C++ math functions in `cmath`. Is there an equivalent C++ function for `np.sign`? If not, we could potentially define our own implementation of `np.sign` in C++ during translation.

Also, what other math functions should we support?

**Comment by tylerjereddy** [comment]
> Also, what other math functions should we support?

I added a few comments about that in gh-28, which drafts in a comparison test. Some things are available in i.e., boost, but a lot of cases you can write in 1-2 lines of C++ like code, so not sure we gain too much in those cases.

The main thing this got me thinking more deeply about is whether you'd want to allow devs/users to avoid writing workloads completely by providing an actual "ufunc" interface in pykokkos--so something like `pk.sqrt(view)` to match `np.sqrt(array)`. It might help adoption, but not sure that's really on the menu in the near-term.

**Comment by NaderAlAwar** [comment]
So far we've added `pk.reciprocal` (#29, #39), `pk.sqrt` (#36), `pk.log` (#37), and `pk.sign` (#45). Should we have a list of remaining ufuncs that we're planning on adding?
